modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/manager/ExtensionRegistry.java

```diff
@@ -50,13 +52,16 @@
  */
 final class ExtensionRegistry {
 
-  private final LoadingCache<ExtensionModel, List<ConfigurationProvider>> providersByExtension =
-      CacheBuilder.newBuilder().build(new CacheLoader<ExtensionModel, List<ConfigurationProvider>>() {
+  private final LoadingCache<ExtensionModel, Multimap<ConfigurationModel, ConfigurationProvider>> providersByExtension =
+      CacheBuilder.newBuilder().build(new CacheLoader<ExtensionModel, Multimap<ConfigurationModel, ConfigurationProvider>>() {
 
         @Override
-        public List<ConfigurationProvider> load(ExtensionModel key) throws Exception {
-          return registry.lookupObjects(ConfigurationProvider.class).stream()
+        public Multimap<ConfigurationModel, ConfigurationProvider> load(ExtensionModel key) throws Exception {
```

Comment:
when do we have more than one configurationProvider for the same ConfigurationModel?